frontend/viewer/src/lib/components/field-editors/select.svelte

```diff
@@ -0,0 +1,164 @@
+    ...constProps
+  }: {
+    value?: Value;
+    options: Readonly<Readonly<Value>[]>;
+    readonly?: boolean;
+    /* eslint-disable @typescript-eslint/no-redundant-type-constituents */
+    idSelector: ConditionalKeys<Value, Primitive> | ((value: Value) => Primitive);
+    labelSelector: ConditionalKeys<Value, string> | ((value: Value) => string);
+    /* eslint-enable @typescript-eslint/no-redundant-type-constituents */
+    placeholder?: string;
+    filterPlaceholder?: string;
+    emptyResultsPlaceholder?: string;
+    drawerTitle?: string;
+  } = $props();
+
+  const {
+    options,
+    readonly = false,
+    idSelector,
+    labelSelector,
+    placeholder,
+    filterPlaceholder,
+    emptyResultsPlaceholder,
+    drawerTitle,
+  } = $derived(constProps);
+
+  const getId = $derived.by(() => {
+    if (typeof idSelector === 'function') return idSelector;
+    return (value: Value) => value[idSelector] as Primitive;
+  });
+
+  const getLabel = $derived.by(() => {
+    if (typeof labelSelector === 'function') return labelSelector;
+    return (value: Value) => value[labelSelector] as string;
+  });
+
+  let open = $state(false);
+  let filterValue = $state('');
+  let triggerRef = $state<HTMLButtonElement | null>(null);
+  let commandRef = $state<HTMLElement | null>(null);
+
+  watch(() => open, () => {
+    filterValue = '';
+  });
+
+  function dismiss() {
+    open = false;
+  }
+
+  function selectValue(newValue: Value) {
+    value = newValue;
+    open = false;
+  }
+
+  const filteredOptions = $derived.by(() => {
+    const filterValueLower = filterValue.toLocaleLowerCase();
+    return options.filter((option) => {
+      const label = getLabel(option).toLocaleLowerCase();
+      return label.includes(filterValueLower);
+    });
+  });
+</script>
+
+{#snippet trigger({ props }: { props: Record<string, unknown> })}
+  <Button disabled={readonly} bind:ref={triggerRef} variant="outline" {...props} role="combobox" aria-expanded={open}
+    class="w-full h-auto px-2 justify-between disabled:opacity-100 disabled:border-transparent">
+    {#if value}
+      <span>
+        {getLabel(value)}
+      </span>
+    {:else}
+      <span class="text-muted-foreground">
+        {placeholder ?? $t`None`}
+        <!-- ensures that baseline alignment works for consumers of this component -->
+        &nbsp;
+      </span>
+    {/if}
+    {#if !readonly}
+      <Icon icon="i-mdi-chevron-down" class="mr-2 size-5 shrink-0 opacity-50" />
+    {/if}
+  </Button>
+{/snippet}
+
+{#snippet command()}
+  <Command shouldFilter={false} bind:ref={commandRef}>
+    <CommandInput bind:value={filterValue} autofocus placeholder={filterPlaceholder ?? $t`Filter...`} {onkeydown}>
+      <div class="flex items-center gap-2 flex-nowrap">
+        {#if IsMobile.value}
+          {#if filterValue}
+            <Button variant="ghost" size="xs-icon" onclick={() => (filterValue = '')} aria-label={$t`clear`}>
+              <Icon icon="i-mdi-close" />
```

Comment:
any reason you're not just setting the icon prop on the button? both here and below?